File: js/linearSearch.js
const arrayToSearchThrough = [];
for (let i = 1; i <= 1000; i++) {
    arrayToSearchThrough.push(i);
}

exports.linearSearch = function(valueToFind, arrayToSearchThrough) {
    for (let i = 0; i < arrayToSearchThrough.length; i++) {
        if (arrayToSearchThrough[i] === valueToFind) return i;
        if (i === arrayToSearchThrough.length-1) return undefined;
    }
};

exports.linearSearchGlobal = function (valueToFind, arrayToSearchThrough) {
    let result = [];
    for (let i = 0; i < arrayToSearchThrough.length; i++) {
        if (arrayToSearchThrough[i] === valueToFind) result.push(i)
        if (i === arrayToSearchThrough.length-1 && result === []) return undefined;
        if (i === arrayToSearchThrough.length-1 && result !== []) return result;
    }
};

File: python/linear_search.py
array_to_search_through = []
for number in range(1, 1001):
    array_to_search_through.append(number)

def linear_search(value_to_find, array_to_search_through):
    for i in range(len(array_to_search_through)):
        if value_to_find == array_to_search_through[i]:
            return i

# print(linear_search(3, [1, 2, 3, 4, 5]))

def global_linear_search(value_to_find, array_to_search_through):
    result = []
    for i in range(len(array_to_search_through)):
        if value_to_find == array_to_search_through[i]:
            result.append(i)
    return result